### PR TITLE
Fix doc links

### PR DESCRIPTION
**Pull request by k-nasa**
## Description

Currently method link is plain text, Because link format is invalid. I fixed it.

スクリーンショット 2019-05-07 10 54 34

スクリーンショット 2019-05-07 10 55 08

## How Has This Been Tested?

I see new documentation.

スクリーンショット 2019-05-07 10 57 15

スクリーンショット 2019-05-07 10 57 10

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/rustasync/tide/blob/master/.github/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/route.rs [hunk 1]
```diff
@@ -2,7 +2,7 @@ use crate::{router::Router, Endpoint};
 
 /// A handle to a route.
 ///
-/// All HTTP requests are made against resources. After using [`App::at`] (or
+/// All HTTP requests are made against resources. After using [`App::at`](crate::app::App::new) (or
 /// [`Route::at`]) to establish a route, the `Route` type can be used to
 /// establish endpoints for various HTTP methods at that path. Also, using
 /// `nest`, it can be used to set up a subrouter.
```
Comment (yoshuawuyts) [inline src/route.rs]:
```suggestion
 /// `nest`, it can be used to set up a subrouter.

[`App::at`]: ../struct.App.html#method.at
```

Comment (k-nasa) [reply src/route.rs]: Why should I change the notation of the path?

File: src/route.rs [hunk 2]
```diff
@@ -2,7 +2,7 @@ use crate::{router::Router, Endpoint};
 
 /// A handle to a route.
 ///
-/// All HTTP requests are made against resources. After using [`App::at`] (or
+/// All HTTP requests are made against resources. After using [`App::at`](crate::app::App::new) (or
```
Comment (yoshuawuyts) [inline src/route.rs]:
```suggestion
/// All HTTP requests are made against resources. After using [`App::at`] (or
```

File: src/route.rs (hunk 1 above)
Comment (yoshuawuyts) [reply src/route.rs]: My understanding was that [`tide::Route`](https://docs.rs/tide/0.2.0/tide/struct.Route.html) wasn't rendering the link to `App::at` correctly.

Using link blocks at the bottom of a paragraph is pretty standard practice in Rust's stdlib, so I that's why I proposed making this change.

Comment (k-nasa) [reply src/route.rs]: Ok.  I did not know that I could include a link block. Thanks 😄

**Comment by k-nasa** [comment]
@yoshuawuyts I fixed it!  Please check when there is time

**Pull request merged**